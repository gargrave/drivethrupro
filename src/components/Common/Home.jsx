import React from 'react'

const HomeRoute = () => {
  return (
    <div>
      <h2>Home</h2>
    </div>
  )
}

export default HomeRoute
