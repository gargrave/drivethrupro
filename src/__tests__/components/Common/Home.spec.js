import React from 'react'
import { shallow } from 'enzyme'

import Home, { Unwrapped } from '../../../components/Common/Home'

describe('Home (without Redux)', () => {
  let props
  let component

  beforeEach(() => {
    props = {
      ordersCompleted: 0,
      ordersCancelled: 0,
      revenueEarned: 0,
      revenueLost: 0
    }
    component = shallow(<Unwrapped {...props} />)
  })

  test('Home renders correctly', () => {
    expect(component).toMatchSnapshot()
  })
})
