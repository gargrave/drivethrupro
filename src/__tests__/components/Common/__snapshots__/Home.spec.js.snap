// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home (without Redux) Home renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h2>
        DriveThru Pro
    </h2>
    <hr />
    <h4>
        Links:
    </h4>
    <h5>
        <Link
            replace={false}
            to="/orders"
        >
            View Open Orders
        </Link>
    </h5>
    <h5>
        <Link
            replace={false}
            to="/orders/new"
        >
            Create a New Order
        </Link>
    </h5>
    <hr />
    <h4>
        Stats:
    </h4>
    <h5>
        You have 
        <span
            style={
                Object {
                    "color": "green",
                  }
            }
        >
            completed
        </span>
         
        <strong>
            0 orders
        </strong>
        , for a total of
         
        <strong>
            $0.00
        </strong>
        .
    </h5>
    <h5>
        You have 
        <span
            style={
                Object {
                    "color": "red",
                  }
            }
        >
            cancelled
        </span>
         
        <strong>
            0 orders
        </strong>
        , for a total loss of
         
        <strong>
            $0.00
        </strong>
        .
    </h5>
</div>,
  "nodes": Array [
    <div>
      <h2>
            DriveThru Pro
      </h2>
      <hr />
      <h4>
            Links:
      </h4>
      <h5>
            <Link
                  replace={false}
                  to="/orders"
            >
                  View Open Orders
            </Link>
      </h5>
      <h5>
            <Link
                  replace={false}
                  to="/orders/new"
            >
                  Create a New Order
            </Link>
      </h5>
      <hr />
      <h4>
            Stats:
      </h4>
      <h5>
            You have 
            <span
                  style={
                        Object {
                              "color": "green",
                            }
                  }
            >
                  completed
            </span>
             
            <strong>
                  0 orders
            </strong>
            , for a total of
             
            <strong>
                  $0.00
            </strong>
            .
      </h5>
      <h5>
            You have 
            <span
                  style={
                        Object {
                              "color": "red",
                            }
                  }
            >
                  cancelled
            </span>
             
            <strong>
                  0 orders
            </strong>
            , for a total loss of
             
            <strong>
                  $0.00
            </strong>
            .
      </h5>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <HomeRoute
        ordersCancelled={0}
        ordersCompleted={0}
        revenueEarned={0}
        revenueLost={0}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "ordersCancelled": 0,
          "ordersCompleted": 0,
          "revenueEarned": 0,
          "revenueLost": 0,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <h2>
                    DriveThru Pro
          </h2>
          <hr />
          <h4>
                    Links:
          </h4>
          <h5>
                    <Link
                              replace={false}
                              to="/orders"
                    >
                              View Open Orders
                    </Link>
          </h5>
          <h5>
                    <Link
                              replace={false}
                              to="/orders/new"
                    >
                              Create a New Order
                    </Link>
          </h5>
          <hr />
          <h4>
                    Stats:
          </h4>
          <h5>
                    You have 
                    <span
                              style={
                                        Object {
                                                  "color": "green",
                                                }
                              }
                    >
                              completed
                    </span>
                     
                    <strong>
                              0 orders
                    </strong>
                    , for a total of
                     
                    <strong>
                              $0.00
                    </strong>
                    .
          </h5>
          <h5>
                    You have 
                    <span
                              style={
                                        Object {
                                                  "color": "red",
                                                }
                              }
                    >
                              cancelled
                    </span>
                     
                    <strong>
                              0 orders
                    </strong>
                    , for a total loss of
                     
                    <strong>
                              $0.00
                    </strong>
                    .
          </h5>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <h2>
                    DriveThru Pro
          </h2>
          <hr />
          <h4>
                    Links:
          </h4>
          <h5>
                    <Link
                              replace={false}
                              to="/orders"
                    >
                              View Open Orders
                    </Link>
          </h5>
          <h5>
                    <Link
                              replace={false}
                              to="/orders/new"
                    >
                              Create a New Order
                    </Link>
          </h5>
          <hr />
          <h4>
                    Stats:
          </h4>
          <h5>
                    You have 
                    <span
                              style={
                                        Object {
                                                  "color": "green",
                                                }
                              }
                    >
                              completed
                    </span>
                     
                    <strong>
                              0 orders
                    </strong>
                    , for a total of
                     
                    <strong>
                              $0.00
                    </strong>
                    .
          </h5>
          <h5>
                    You have 
                    <span
                              style={
                                        Object {
                                                  "color": "red",
                                                }
                              }
                    >
                              cancelled
                    </span>
                     
                    <strong>
                              0 orders
                    </strong>
                    , for a total loss of
                     
                    <strong>
                              $0.00
                    </strong>
                    .
          </h5>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <HomeRoute
    ordersCancelled={0}
    ordersCompleted={0}
    revenueEarned={0}
    revenueLost={0}
/>,
}
`;
